### ISSUE TITLE
WIX0103 Cannot Find File Error in wixlib project

### ISSUE THREAD

**Issue by weberms** [wixtoolset/issues#8734]
### WiX Version

5.0.1

### .NET or MSBuild or Visual Studio Version

MSBuild version 17.8.3+195e7f5a3 for .NET Framework

### HeatWave Version

1.0.4.5

### Windows Version

W2K16 Server 14393.7159

### Repro Steps

1. Create a new wixlib project
2. Add following code
`<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
	<Fragment>
        <DirectoryRef Id="INSTALLFOLDER">
            <Component Id="Test_Files" Guid="{8FCF669B-612E-4E4F-B9C1-8DBC2D8D2EB6}" Shared="yes" Bitness="always64">
                <File Id="File1.xml"  Source="..\Config Files\XML\File1.xml" Vital="yes" />
                <File Id="File2.xml" Source="..\Config Files\XML\File2.xml" Vital="yes" />
            </Component>
        </DirectoryRef>
	</Fragment>
</Wix>`

3.  Create another wixlib project and reference the wixlib project with the xml files.
`<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">    
    <Fragment>
        <ComponentGroup Id="Test">
            <ComponentRef Id="Test_Files" />
        </ComponentGroup>
    </Fragment>`

4.  Build the project in step #3.  

error WIX0103: Cannot find the File file 'wix-ir/File1.xml'. The following paths were checked: wix-ir/File1.xml....

### Actual Result

error WIX0103: Cannot find the File file 'wix-ir/File1.xml'. The following paths were checked: wix-ir/File1.xml....

All the mentioned paths searched for the file include an invalid "wix-ir" folder.

### Expected Result

project builds successfully and includes the specified files like it does in WIX 3.11.

### Acknowledgements

- [X] I acknowledge that this is a fully completed bug report. It is not a question or attempt to get help debugging my issue (because those should be sent to Discussions).

**Comment by weberms** [comment]
It seems there was a breaking change to how wixlib projects are handled in Wix 4+.  We use wixlib projects extensively to centralize and share common bits needed by multiple installers but we're seeing the above error as well as seeming related error "WIX0001: System.IO.DirectoryNotFoundException: Could not find a part of the path....".  These projects have worked reliably for years in v3.11.  Only change was upgrade to Wix 4+

**Comment by barnson** [comment]
Please provide a [repro repo](https://www.joyofsetup.com/2024/07/30/force-multipliers/).

**Comment by weberms** [comment]
> Please provide a [repro repo](https://www.joyofsetup.com/2024/07/30/force-multipliers/).

I haven't been able to reproduce the issue yet but I cannot share the exact code and it's difficult to duplicate the complex layering and functionality we have in our installers without the original code.

**Comment by weberms** [comment]
@barnson were you able to repo with the sample code?

**Comment by weberms** [comment]
Any update on this?  This is a hard blocker for our move to Wix 4.

**Comment by barnson** [comment]
No, there is no update. If you require an SLA, you [should become a customer](https://www.firegiant.com/services/). Otherwise, you need to wait for the issue to be triaged and for someone to decide to volunteer the time to investigate and fix the issue.

**Comment by SOHODeveloper** [comment]
@weberms 
I too am experiencing this issue.  Have you found a workaround etc?

**Comment by weberms** [comment]
I have not found a workaround and this continues to be an issue in Wix 6.

**Comment by weberms** [comment]
@barnson or @robmen is this a bug in Wix 4+ or a scenario that is no longer supported?  We have been waiting to migrate off of Wix 3 due to this problem but if its not going to be fixed we will consider refactoring our MSIs.  That is not ideal for sure because its a lot of work and will require adding duplicate code to any MSI that relies on the common functionality in the wixlib projects.

**Comment by robmen** [comment]
The issue is `open`, `unassigned`, and `up for grabs`.  That means triage thinks it's probably an issue, but it is waiting for someone to take an interest in fixing it. If you are interested, leave a comment, and triage will assign it to you in the next reasonable milestone (but it is `up for grabs` so you can feel pretty confident in starting a fix 😄 ). There is a bit over a month left to get it into [WiX v7](https://github.com/wixtoolset/issues/milestone/31).
